File: 0125-valid-palindrome/0125-valid-palindrome.cs
public class Solution {
    public bool IsPalindrome(string s) {
        StringBuilder sb = new StringBuilder();
        foreach(char c in s.ToLower()){
            if(('a' <= c && c <= 'z') || ('0' <= c && c <= '9')){
                sb.Append(c);
            }
        }
        s = sb.ToString();
        //Console.WriteLine(sb.ToString());
        int left = 0;
        int right = s.Length - 1;
        while(left < right){
            if(s[left] != s[right]){
                return false;
            }
            left++;
            right--;
        }        
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string str = "";
        for(int i = 0; i < s.length(); i++){
            if('A' <= s[i] && s[i] <= 'Z'){
                str += tolower(s[i]);
            }else if(('a' <= s[i] && s[i] <= 'z') || ('0' <= s[i] && s[i] <= '9')){
                str += s[i];
            }
        }
        int left = 0;
        int right = str.length() - 1;
        while(left < right){
            if(str[left] != str[right]){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
};